File: python/bubble-sort.py
def bubble_sort(T):
    """Sorts a list using bubble sort.
    :param a: List to sort
    :type a: list

    :returns: Sorted list
    :rtype: list"""
    ordered = False
    while not ordered:
        ordered = True
        for i in range(0, len(T)-1):
            if T[i+1] < T[i]:
                temp = T[i+1]
                T[i+1] = T[i]
                T[i] = temp
                ordered = False
    return T

File: ruby/bubble-sort.rb
# frozen_string_literal: true

# Sorts a list using bubble sort alogithm
#
# @param t [Hash] elements to sort
def bubble_sort(t)
  ordered = false
  until ordered
    ordered = true
    (0..t.length - 2).each do |i|
      next unless t[i] > t[i + 1]

      t[i], t[i + 1] = t[i + 1], t[i]
      ordered = false
    end
  end
  t
end
